aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -0,0 +1,342 @@
+   */
+  public static final String CLIENT_ASSUME_ROLE_EXTERNAL_ID = "client.assume-role.external-id";
+
+  /**
+   * Used by {@link AssumeRoleAwsClientFactory}.
+   * If set, all AWS clients except STS client will use the given region instead of the default region chain.
+   * <p>
+   * The value must be one of {@link software.amazon.awssdk.regions.Region}, such as 'us-east-1'.
+   * For more details, see https://docs.aws.amazon.com/general/latest/gr/rande.html
+   */
+  public static final String CLIENT_ASSUME_ROLE_REGION = "client.assume-role.region";
+
+  private String s3FileIoSseType;
+  private String s3FileIoSseKey;
+  private String s3FileIoSseMd5;
+  private int s3FileIoMultipartUploadThreads;
+  private int s3FileIoMultiPartSize;
+  private double s3FileIoMultipartThresholdFactor;
+  private String s3fileIoStagingDirectory;
+  private ObjectCannedACL s3FileIoAcl;
+
+  private String glueCatalogId;
+  private boolean glueCatalogSkipArchive;
+
+  private String dynamoDbTableName;
+
+  public AwsProperties() {
+    this.s3FileIoSseType = S3FILEIO_SSE_TYPE_NONE;
+    this.s3FileIoSseKey = null;
+    this.s3FileIoSseMd5 = null;
+    this.s3FileIoAcl = null;
+
+    this.s3FileIoMultipartUploadThreads = Runtime.getRuntime().availableProcessors();
+    this.s3FileIoMultiPartSize = S3FILEIO_MULTIPART_SIZE_DEFAULT;
+    this.s3FileIoMultipartThresholdFactor = S3FILEIO_MULTIPART_THRESHOLD_FACTOR_DEFAULT;
+    this.s3fileIoStagingDirectory = System.getProperty("java.io.tmpdir");
+
+    this.glueCatalogId = null;
+    this.glueCatalogSkipArchive = GLUE_CATALOG_SKIP_ARCHIVE_DEFAULT;
+
+    this.dynamoDbTableName = DYNAMODB_TABLE_NAME_DEFAULT;
+  }
+
+  public AwsProperties(Map<String, String> properties) {
+    this.s3FileIoSseType = properties.getOrDefault(
+        AwsProperties.S3FILEIO_SSE_TYPE, AwsProperties.S3FILEIO_SSE_TYPE_NONE);
+    this.s3FileIoSseKey = properties.get(AwsProperties.S3FILEIO_SSE_KEY);
+    this.s3FileIoSseMd5 = properties.get(AwsProperties.S3FILEIO_SSE_MD5);
+    if (AwsProperties.S3FILEIO_SSE_TYPE_CUSTOM.equals(s3FileIoSseType)) {
+      Preconditions.checkNotNull(s3FileIoSseKey, "Cannot initialize SSE-C S3FileIO with null encryption key");
+      Preconditions.checkNotNull(s3FileIoSseMd5, "Cannot initialize SSE-C S3FileIO with null encryption key MD5");
+    }
+
+    this.glueCatalogId = properties.get(GLUE_CATALOG_ID);
+    this.glueCatalogSkipArchive = PropertyUtil.propertyAsBoolean(properties,
+        AwsProperties.GLUE_CATALOG_SKIP_ARCHIVE, AwsProperties.GLUE_CATALOG_SKIP_ARCHIVE_DEFAULT);
+
+    this.s3FileIoMultipartUploadThreads = PropertyUtil.propertyAsInt(properties, S3FILEIO_MULTIPART_UPLOAD_THREADS,
+        Runtime.getRuntime().availableProcessors());
+
+    try {
+      this.s3FileIoMultiPartSize = PropertyUtil.propertyAsInt(properties, S3FILEIO_MULTIPART_SIZE,
+          S3FILEIO_MULTIPART_SIZE_DEFAULT);
+    } catch (NumberFormatException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)